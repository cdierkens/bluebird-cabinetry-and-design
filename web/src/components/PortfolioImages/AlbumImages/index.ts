export { default as AlbumImages } from "./AlbumImages";
