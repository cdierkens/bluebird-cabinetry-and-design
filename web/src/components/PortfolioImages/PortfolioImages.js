import { graphql, useStaticQuery } from "gatsby";
import PropTypes from "prop-types";
import React, { useMemo, useRef, useState } from "react";
import Carousel, { Modal, ModalGateway } from "react-images";
import resolveConfig from "tailwindcss/resolveConfig";
import useQueryString from "use-query-string";
import tailwindConfig from "../../../tailwind.config.js";
import { LeftArrowIcon, RightArrowIcon } from "../../icons";
import { builder } from "../../lib/image-url";
import Container from "../container";
import Tags from "./Tags/Tags.js";

const { theme } = resolveConfig(tailwindConfig);

export const portfolioImagesQuery = graphql`
  query PortfolioImages {
    images: allSanityPortfolioImage {
      nodes {
        image {
          description
          file {
            asset {
              id
            }
          }
        }
        caption
        tags
      }
    }
  }
`;

const navButtonStyles = (base) => ({
  ...base,
  backgroundColor: theme.colors.white,
  boxShadow: theme.boxShadow.md,
  color: theme.colors.gray.darker,

  "&:hover, &:active, &:active": {
    backgroundColor: theme.colors.white,
    color: theme.colors.black,
  },
  "&:focus": {
    boxShadow: theme.boxShadow.outline,
    outline: 0,
  },
});

const pageSize = 9;

const PortfolioImages = ({ location }) => {
  const {
    images: { nodes: portfolioImages },
  } = useStaticQuery(portfolioImagesQuery);

  const [selectedIndex, setSelectedIndex] = useState(0);
  const [lightboxIsOpen, setLightboxIsOpen] = useState(false);

  const [query, setQuery] = useQueryString(location, (value) => {
    if (typeof window === "object") {
      window.history.pushState(null, window.document.title, value);
    }
  });

  const tags = useMemo(
    () =>
      Array.from(
        portfolioImages.reduce(
          (keys, node) => new Set([...node.tags, ...keys]),
          new Set()
        )
      ).sort(),
    [portfolioImages]
  );

  const grid = useRef();

  const checkedTags = query.tags ? decodeURI(query.tags).split(",") : tags;
  const filteredImages = portfolioImages.filter(({ tags }) =>
    tags.find((tag) => checkedTags.find((checkedTag) => checkedTag === tag))
  );
  const selectedPage = Math.floor(selectedIndex / pageSize);

  const images = filteredImages.map(({ image, caption }) => ({
    source: {
      download: builder.image(image.file.asset.id).url(),
      fullscreen: builder
        .image(image.file.asset.id)
        .height(1080)
        .fit("clip")
        .url(),
      regular: builder.image(image.file.asset.id).height(900).fit("clip").url(),
      thumbnail: builder
        .image(image.file.asset.id)
        .height(400)
        .fit("clip")
        .url(),
    },
    caption,
    alt: image.description,
  }));

  const openLightbox = (index) => {
    setSelectedIndex(index);
    setLightboxIsOpen(true);
  };

  const closeLightbox = () => {
    setLightboxIsOpen(false);
  };

  const handlePagination = (index) => {
    if (index === selectedIndex) {
      return;
    }

    if (grid.current) {
      grid.current.scrollIntoView({ behavior: "smooth" });
    }
    setSelectedIndex(index);
  };

  return (
    <Container className="my-6">
      <h2>Photos</h2>

      <Tags
        tags={tags.map((tag) => ({
          value: tag,
          checked: Boolean(
            checkedTags.find((checkedTag) => checkedTag === tag)
          ),
          onChange: ({ target }) => {
            if (target.checked) {
              setQuery({
                tags: [...checkedTags.map(encodeURI), encodeURI(tag)].join(","),
              });
            } else {
              setQuery({
                tags: checkedTags
                  .filter((checkedTags) => checkedTags !== tag)
                  .map(encodeURI)
                  .join(","),
              });
            }
            setSelectedIndex(0);
          },
        }))}
      />

      <div className="p-1">
        <button
          className="text-blue-dark hover:text-gold text-base m-2"
          onClick={() => setQuery({ tags: tags.map(encodeURI).join(",") })}
        >
          Show All
        </button>
        |
        <button
          className="text-blue-dark hover:text-gold text-base m-2"
          onClick={() => setQuery({ tags: [] })}
        >
          Show None
        </button>
      </div>

      <>
        {filteredImages.map(({ image }) => (
          <img
            key={image.file.asset.id}
            className="sr-only"
            src={builder
              .image(image.file.asset.id)
              .size(408, 272)
              .fit("crop")
              .url()}
            alt={image.description}
          />
        ))}
      </>

      <div className="grid grid-cols-1 sm:grid-cols-3 gap-3 pt-6" ref={grid}>
        {filteredImages
          .slice(pageSize * selectedPage, pageSize * selectedPage + pageSize)
          .map(({ image }, index) => (
            <div key={image.file.asset.id}>
              <button
                className="transform hover:scale-105 focus:scale-105 duration-300 p-1 shadow-md focus:outline-none focus:ring bg-white"
                onClick={() => openLightbox(index + pageSize * selectedPage)}
              >
                <img
                  className="pointer-events-none"
                  src={builder
                    .image(image.file.asset.id)
                    .size(408, 272)
                    .fit("crop")
                    .url()}
                  alt={image.description}
                />
              </button>
            </div>
          ))}
      </div>

      {filteredImages.length > pageSize ? (
        <div className="flex items-center justify-center mt-2">
          <button
            onClick={() =>
              handlePagination(Math.max(selectedIndex - pageSize, 0))
            }
            className={`bg-blue-dark text-white rounded-full inline-block w-8 h-8 mr-3 hover:bg-gold cursor-pointer focus:outline-none focus:ring`}
          >
            <LeftArrowIcon className="w-full h-full p-2" />
            <span className="sr-only">Previous Page</span>
          </button>

          {Array.from({
            length: Math.floor(filteredImages.length / pageSize) + 1,
          }).map((_, index) => (
            <button
              key={index}
              onClick={() => handlePagination(index * pageSize)}
              className={`text-white rounded-full text-center font-medium p-1 inline-block w-8 h-8 mr-3 hover:bg-gold cursor-pointer focus:outline-none focus:ring ${
                index === selectedPage ? "bg-gold" : "bg-blue-dark"
              }`}
            >
              {index + 1}
            </button>
          ))}

          <button
            onClick={() =>
              handlePagination(
                Math.min(
                  selectedIndex + pageSize,
                  Math.floor(filteredImages.length / pageSize) * pageSize
                )
              )
            }
            className={`bg-blue-dark text-white rounded-full inline-block w-8 h-8 mr-3 hover:bg-gold cursor-pointer focus:outline-none focus:ring`}
          >
            <RightArrowIcon className="w-full h-full p-2" />
            <span className="sr-only">Next Page</span>
          </button>
        </div>
      ) : null}

      <ModalGateway>
        {lightboxIsOpen ? (
          <Modal
            closeOnBackdropClick={false}
            onClose={closeLightbox}
            styles={{
              blanket: (base) => ({
                ...base,
                backgroundColor: theme.colors.gray.light,
              }),
              positioner: (base) => ({
                ...base,
                display: "block",
              }),
            }}
          >
            <Carousel
              hideControlsWhenIdle={false}
              currentIndex={selectedIndex}
              components={{ Header, View, Footer }}
              views={images}
              trackProps={{
                onViewChange: setSelectedIndex,
              }}
              styles={customStyles}
            />
          </Modal>
        ) : null}
      </ModalGateway>
    </Container>
  );
};

const customStyles = {
  container: (base) => ({
    ...base,
    height: "100vh",
  }),
  navigationPrev: navButtonStyles,
  navigationNext: navButtonStyles,
};

PortfolioImages.propTypes = {
  location: PropTypes.object.isRequired,
};

const Footer = ({ currentIndex, views, modalProps: { isFullscreen } }) => {
  return (
    <div className="absolute bottom-2 right-2 text-right px-2 py-1 text-sm rounded bg-gray-light">
      <span className="sr-only">Image number</span> {currentIndex + 1} of{" "}
      {views.length}
    </div>
  );
};

Footer.propTypes = {
  currentIndex: PropTypes.object.isRequired,
  views: PropTypes.array.isRequired,
  currentView: PropTypes.object.isRequired,
  modalProps: PropTypes.object.isRequired,
};

const View = ({ views, index, modalProps: { isFullscreen } }) => {
  const { source, alt } = views[index];

  return (
    <div
      className="leading-none relative text-center box-border flex items-center justify-center"
      style={{ height: isFullscreen ? "100vh" : "calc(100vh - 90px)" }}
    >
      <img
        className="h-auto max-h-full max-w-full m-auto select-none"
        src={source.regular}
        alt={alt}
      />
    </div>
  );
};

View.propTypes = {
  views: PropTypes.array,
  index: PropTypes.number,
  modalProps: PropTypes.object.isRequired,
};

const Header = ({
  currentView: { caption },
  modalProps: { onClose, toggleFullscreen, isFullscreen },
}) => {
  if (isFullscreen) {
    return (
      <button
        onClick={toggleFullscreen}
        className="fixed z-50 top-0 right-0 inline-block p-4 m-2 text-gray-light hover:text-gold focus:outline-none focus:ring bg-black bg-opacity-25 rounded"
      >
        <span className="sr-only">Close</span>
        <svg
          role="presentation"
          viewBox="0 0 24 24"
          className="h-6 w-6 fill-current stroke-current"
        >
          <path d="M18.984 6.422l-5.578 5.578 5.578 5.578-1.406 1.406-5.578-5.578-5.578 5.578-1.406-1.406 5.578-5.578-5.578-5.578 1.406-1.406 5.578 5.578 5.578-5.578z"></path>
        </svg>
      </button>
    );
  }

  return (
    <div className="p-1 md:p-4 bg-gray-light shadow-sm flex justify-between items-center">
      <div className="truncate" style={{ maxWidth: "calc(100% - 80px)" }}>
        <span className="px-1 whitespace-nowrap truncate">{caption}</span>
      </div>
      <div>
        <button
          onClick={toggleFullscreen}
          className="inline-block p-2 hover:text-black text-gray-darker focus:outline-none focus:ring"
        >
          <span className="sr-only">Fullscreen</span>
          <svg
            className="h-6 w-6 fill-current stroke-current"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path d="M14.016 5.016h4.969v4.969h-1.969v-3h-3v-1.969zM17.016 17.016v-3h1.969v4.969h-4.969v-1.969h3zM5.016 9.984v-4.969h4.969v1.969h-3v3h-1.969zM6.984 14.016v3h3v1.969h-4.969v-4.969h1.969z"></path>
          </svg>
        </button>
        <button
          onClick={onClose}
          className="inline-block p-2 hover:text-black text-gray-darker focus:outline-none focus:ring"
        >
          <span className="sr-only">Close</span>
          <svg
            role="presentation"
            viewBox="0 0 24 24"
            className="h-6 w-6 fill-current stroke-current"
          >
            <path d="M18.984 6.422l-5.578 5.578 5.578 5.578-1.406 1.406-5.578-5.578-5.578 5.578-1.406-1.406 5.578-5.578-5.578-5.578 1.406-1.406 5.578 5.578 5.578-5.578z"></path>
          </svg>
        </button>
      </div>
    </div>
  );
};

Header.propTypes = {
  currentView: PropTypes.object.isRequired,
  modalProps: PropTypes.object.isRequired,
};

export default PortfolioImages;
