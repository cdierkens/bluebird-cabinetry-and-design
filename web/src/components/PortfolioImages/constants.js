import resolveConfig from "tailwindcss/resolveConfig";
import tailwindConfig from "../../../tailwind.config.js";

export const PAGE_SIZE = 9;
export const ROOM_NAMES = ["Kitchen", "Bath", "Laundry", "Office"];
export const LABEL_NAMES = ["Banquette", "Bar", "Island", "3D Renderings"];

export const { theme } = resolveConfig(tailwindConfig);
