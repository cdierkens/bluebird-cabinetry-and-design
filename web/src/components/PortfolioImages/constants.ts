import resolveConfig from "tailwindcss/resolveConfig";
import { TailwindConfig } from "tailwindcss/tailwind-config";
import tailwindConfig from "../../../tailwind.config.js";
import { todo } from "../../migration.types.js";

export const PAGE_SIZE = 9;

export const { theme } = resolveConfig(
  (tailwindConfig as todo) as TailwindConfig
);
