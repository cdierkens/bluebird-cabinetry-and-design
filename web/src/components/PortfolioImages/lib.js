import { builder } from "../../lib/image-url";

export const mapPortfolioImageToCarouselImage = ({
  image,
  title,
  contractor,
  decorator,
  furnitureRefinishing,
  interiorDesigner,
  software,
}) => ({
  source: {
    download: builder.image(image.file.asset.id).url(),
    fullscreen: builder
      .image(image.file.asset.id)
      .height(1080)
      .fit("clip")
      .url(),
    regular: builder.image(image.file.asset.id).height(900).fit("clip").url(),
    thumbnail: builder.image(image.file.asset.id).height(400).fit("clip").url(),
  },
  title,
  contractor,
  decorator,
  furnitureRefinishing,
  interiorDesigner,
  software,
  alt: image.description,
});
