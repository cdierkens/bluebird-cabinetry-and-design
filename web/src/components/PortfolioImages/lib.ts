import { OptionsType } from "react-select";
import { builder } from "../../lib/image-url";
import { todo } from "../../migration.types";

interface GetValuesForAttributeOptions {
  attr: string;
  allImages: todo[];
}

export const getValuesForAttribute = ({
  attr,
  allImages,
}: GetValuesForAttributeOptions) => {
  return Array.from(
    allImages.reduce((keys: Set<string>, node: todo) => {
      const prop = node[attr];
      if (Array.isArray(prop)) {
        prop.forEach((value) => keys.add(value));
      } else {
        keys.add(prop);
      }

      return keys;
    }, new Set())
  )
    .filter(Boolean)
    .sort();
};

export const mapPortfolioImageToCarouselImage = ({
  image,
  title,
  contractor,
  decorator,
  furnitureRefinishing,
  interiorDesigner,
  software,
}: todo) => ({
  source: {
    download: builder.image(image.file.asset.id).url(),
    fullscreen: builder
      .image(image.file.asset.id)
      .height(1080)
      .fit("clip")
      .url(),
    regular: builder.image(image.file.asset.id).height(900).fit("clip").url(),
    thumbnail: builder.image(image.file.asset.id).height(400).fit("clip").url(),
  },
  title,
  contractor,
  decorator,
  furnitureRefinishing,
  interiorDesigner,
  software,
  alt: image.description,
});

export const getArrayFromQueryParam = (
  value: string | string[] | null
): string[] => {
  if (Array.isArray(value)) {
    return value;
  }

  switch (typeof value) {
    case "string":
      return decodeURI(value).split(",").sort();
    default:
      return [];
  }
};

export const getQueryStringFromOption = (values: OptionsType<any>): string => {
  return values
    .map(({ value }) => value)
    .sort()
    .join(",");
};
