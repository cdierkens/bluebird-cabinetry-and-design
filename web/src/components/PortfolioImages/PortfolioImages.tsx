import { graphql, navigate, useStaticQuery } from "gatsby";
import React, { useEffect, useMemo, useRef, useState } from "react";
import {
  MdFilterList,
  MdKeyboardArrowDown,
  MdKeyboardArrowUp,
} from "react-icons/md";
import useQueryString from "use-query-string";
import { PortfolioImagesQuery } from "../../../graphql-types";
import { isDefined } from "../../lib";
import { invariant } from "../../lib/invariant";
import Button from "../Button";
import Container from "../Container";
import { AlbumImages } from "./AlbumImages";
import { PAGE_SIZE } from "./constants";
import { ImageGrid } from "./ImageGrid";
import { ImageModal } from "./ImageModal";
import {
  getArrayFromQueryParam,
  getValuesForAttribute,
  mapPortfolioImageToCarouselImage,
} from "./lib";
import { Pagination } from "./Pagination/Pagination";
import { SelectInput } from "./SelectInput";

interface PortfolioImagesProps {
  location: Location;
}

const PortfolioImages: React.FC<PortfolioImagesProps> = ({ location }) => {
  const {
    allSanityPortfolioImage: { nodes: allImages },
    allSanityAlbum: { nodes: albums },
  } = useStaticQuery<PortfolioImagesQuery>(portfolioImagesQuery);

  // Note: Library author typed this as an interface instead of a tuple.
  const { 0: query, 1: setQuery } = useQueryString(location, navigate);

  // Show all labels, cabinetry, rooms, and finishes for an empty query string.
  useEffect(() => {
    if (query.labels || query.cabinetry || query.finish || query.room) {
      return;
    }

    setQuery({
      ...query,
      labels: getValuesForAttribute({ allImages, attr: "labels" }),
      cabinetry: getValuesForAttribute({ allImages, attr: "cabinetry" }),
      finish: getValuesForAttribute({ allImages, attr: "finish" }),
      room: getValuesForAttribute({ allImages, attr: "room" }),
    });
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const selectedLabels = useMemo(() => {
    return getArrayFromQueryParam(query.labels);
  }, [query.labels]);

  const selectedCabinetry = useMemo(() => {
    return getArrayFromQueryParam(query.cabinetry);
  }, [query.cabinetry]);

  const selectedFinish = useMemo(() => {
    return getArrayFromQueryParam(query.finish);
  }, [query.finish]);

  const selectedRooms = useMemo(() => {
    return getArrayFromQueryParam(query.room);
  }, [query.room]);

  const selectedImages = useMemo(() => {
    return allImages.filter(({ labels, room, cabinetry, finish }) => {
      return (
        selectedLabels.find((selected) =>
          labels?.find((value) => value === selected)
        ) ||
        selectedRooms.find((selected) => selected === room) ||
        selectedCabinetry.find((selected) =>
          cabinetry?.find((value) => value === selected)
        ) ||
        selectedFinish.find((selected) =>
          finish?.find((value) => value === selected)
        )
      );
    });
  }, [
    selectedLabels,
    selectedCabinetry,
    selectedFinish,
    selectedRooms,
    allImages,
  ]);

  useEffect(() => {
    setSelectedIndex(0);
  }, [selectedImages]);

  const carouselImages = selectedImages.map(mapPortfolioImageToCarouselImage);

  const [selectedIndex, setSelectedIndex] = useState(0);
  const [lightboxIsOpen, setLightboxIsOpen] = useState(false);

  const selectedPage = Math.floor(selectedIndex / PAGE_SIZE);
  const scrollToRef = useRef<HTMLHeadingElement>(null);

  const openLightbox = (index: number) => {
    setSelectedIndex(index);
    setLightboxIsOpen(true);
  };

  const closeLightbox = () => {
    setLightboxIsOpen(false);
  };

  const handlePagination = (index: number) => {
    if (index === selectedIndex) {
      return;
    }

    if (scrollToRef.current) {
      scrollToRef.current.scrollIntoView({ behavior: "smooth" });
    }

    setSelectedIndex(index);
  };

  const [showFilters, setShowFilters] = useState(false);

  return (
    <>
      <Container className="my-6">
        <h2 ref={scrollToRef}>All Photos</h2>

        <div className="shadow-md p-3 mb-3">
          <Button
            className="w-full text-left text-blue-dark mb-3 flex justify-between items-center px-2 border border-opacity-50 border-gray-dark"
            onClick={() => setShowFilters((value) => !value)}
          >
            <span>
              <MdFilterList className="inline-block mr-2 h-5 w-5" />
              Filters
            </span>

            {showFilters ? (
              <MdKeyboardArrowUp className="h-6 w-6" />
            ) : (
              <MdKeyboardArrowDown className="h-6 w-6" />
            )}
          </Button>
          {showFilters ? (
            <div>
              <SelectInput
                allImages={allImages}
                attribute="room"
                query={query}
                selectedValues={selectedRooms}
                setQuery={setQuery}
              />

              <SelectInput
                allImages={allImages}
                attribute="labels"
                label="tags"
                query={query}
                selectedValues={selectedLabels}
                setQuery={setQuery}
              />

              <SelectInput
                allImages={allImages}
                attribute="cabinetry"
                query={query}
                selectedValues={selectedCabinetry}
                setQuery={setQuery}
              />

              <SelectInput
                allImages={allImages}
                attribute="finish"
                query={query}
                selectedValues={selectedFinish}
                setQuery={setQuery}
              />
            </div>
          ) : null}
        </div>

        <ImageGrid
          images={selectedImages}
          onClick={(index: number) =>
            openLightbox(index + PAGE_SIZE * selectedPage)
          }
          selectedPage={selectedPage}
        />

        <Pagination
          handlePagination={handlePagination}
          selectedIndex={selectedIndex}
          images={selectedImages}
        />

        <ImageModal
          lightboxIsOpen={lightboxIsOpen}
          closeLightbox={closeLightbox}
          selectedIndex={selectedIndex}
          carouselImages={carouselImages}
          setSelectedIndex={setSelectedIndex}
        />
      </Container>

      {albums.length ? (
        <Container>
          <h2>Project Photos</h2>
          {albums.map(({ title, images }) => {
            invariant(title);
            invariant(images);

            return (
              <AlbumImages
                key={title}
                images={images.filter(isDefined)}
                title={title}
              />
            );
          })}
        </Container>
      ) : null}
    </>
  );
};

export const portfolioImagesQuery = graphql`
  fragment Image on SanityPortfolioImage {
    cabinetry
    caption
    contractor
    decorator
    finish
    furnitureRefinishing
    interiorDesigner
    labels
    room
    software
    title
    image {
      description
      file {
        asset {
          id
          gatsbyImageData(
            height: 360
            width: 640
            fit: CROP
            placeholder: BLURRED
          )
        }
      }
    }
  }
  query PortfolioImages {
    allSanityPortfolioImage {
      nodes {
        ...Image
      }
    }
    allSanityAlbum {
      nodes {
        images {
          ...Image
        }
        title
      }
    }
  }
`;

export default PortfolioImages;
