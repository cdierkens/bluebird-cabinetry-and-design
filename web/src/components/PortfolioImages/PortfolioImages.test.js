import { render } from "@testing-library/react";
import React from "react";
import PortfolioImages from "./PortfolioImages";

it("matches snapshot", () => {
  const { asFragment } = render(<PortfolioImages />);

  expect(asFragment()).toMatchSnapshot();
});
