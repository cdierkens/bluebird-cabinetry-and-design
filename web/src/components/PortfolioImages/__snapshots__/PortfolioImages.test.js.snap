// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container my-6"
  >
    <h2>
      Photos
    </h2>
    <span
      class="block text-blue-dark text-base ml-1 mb-1"
    >
      Tags
    </span>
    <div
      class="p-1 shadow-md flex flex-wrap justify-start"
    />
    <div
      class="p-1"
    >
      <button
        class="inline block text-blue-dark hover:text-gold text-base m-2"
      >
        Show All
      </button>
      |
      <button
        class="inline block text-blue-dark hover:text-gold text-base m-2"
      >
        Show None
      </button>
    </div>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-3 pt-6"
    />
  </div>
</DocumentFragment>
`;
